docs/architecture.md

```diff
@@ -3,7 +3,24 @@
 
 While we prefer in-code and live documentation over wiki pages, these are best suited to single services and we don't have a place to document broader processes that involve the interaction of several services. This section aims to provide that overview of how various services fit together.
 
-## Supporter data flow
+## Supporter Events Service
+
+The service aims to propagate [supporter identities](https://github.com/comicrelief/data-models/tree/master/src/schemas/user-service/models/requests) to User Service and [supporter events](https://github.com/comicrelief/data-models/tree/master/src/schemas/erp/models/actions) to ERP Service.
+
+User Service and ERP Service exist on one region only because of their structure; should they be unavailable for any reason, propagation of data from upstream services may be interrupted. Supporter Events is a multi-region service designed to handle incoming traffic from all sources. Events are queued to SQS and processed one by one.
+
+When identities are supplied to the `user-service` section of the payload, Supporter Events returns a salted hash of the identity that can be used by upstream services to identity resources, so that anonymisation won't break relationships between data and services.
+
+Supporter Service is supposed to handle only write requests to User Service and ERP Service. If an upstream service requires data stored in either service, it should either:
```

Comment:
Do you have a preferred naming style for services? I'm happy with Title Case service names that include the word "Service". There are a couple of deviations here though: "Supporter Events" and "Supporter Service" should both be "Supporter Event Service" by this convention.

Also, the repo name uses "Event" in singular, which I think sounds best (just as we have a User Service rather than a Users Service). I noticed I made this mistake too in the diagram... shall I push a commit to fix that, or are you happy to include it in one of your fixups?

Reply:
I will fix it, cheers! 🙂

Reply:
re: preferred names. I would vote for backticked repo names, i.e. `serverless-user-service` and `serverless-erp-service`. A bit verbose, but this way we know exactly what we are talking about.

Reply:
Thanks!

Reply:
858352228cbf9e2e6cd3cfdc7fd381cd164ccf73